### PR TITLE
node: Make sure seed value in candidate block is valid

### LINKED ISSUE #1405: Make sure that the seed is signed by the block generator
**Describe the bug**
Both the Block Acceptor and Validator are responsible for verifying that the candidate block possesses a seed value signed by the block generator. Failing to conduct this verification could allow adversaries to potentially manipulate the score_value in sortition.

Verification:

**block_generator_public_key.verify( signature, message )**
        where `message` is most_recent block.header.seed,
       `signature` is candidate.header.seed and
      `block_generator_public_key`  is `candidate.header.generator_bls_pubkey`

### PR DESCRIPTION
**Pull request by goshawk-3**
fixes #1405

### CODE REVIEW

**Review by herr-seppia — APPROVED** [review]
LGTM

File: node/src/chain/header_validation.rs [hunk 1]
```diff
@@ -99,6 +106,32 @@ impl<'a, DB: database::DB> Validator<'a, DB> {
             Ok(())
         })?;
 
+        // Verify seed field
+        if candidate_block.height > 1 {
```
Comment (fed-franz) [inline node/src/chain/header_validation.rs:110]: Why do we start with height 2?

Comment (goshawk-3) [reply node/src/chain/header_validation.rs:110]: Because `seed` in Genesis block is always empty.

**Review by herr-seppia — APPROVED** [review]
LGTM

File: node/src/chain/header_validation.rs (hunk 1 above)
Comment (goshawk-3) [reply node/src/chain/header_validation.rs]: Corrected

**Review by herr-seppia — APPROVED** [review]

**Pull request merged**